.github/workflows/deploy_tre_nightly.yml

```diff
@@ -0,0 +1,53 @@
+---
+name: Deploy Azure TRE
+# This workflow is the integration build run for pushes to the main branch
+# It also runs on a schedule, serving as the nightly build
+
+on:  # yamllint disable-line rule:truthy
+  schedule:
+    # 1am each night https://crontab.guru/#0_1_*_*_*
+    - cron: "0 1 * * *"
+  workflow_dispatch:
+
+# This will prevent multiple runs of this entire workflow.
+# We should NOT cancel in progress runs as that can destabilize the environment.
+concurrency: "${{ github.workflow }}-${{ github.ref }}"
+
+jobs:
+  run-deploy-tre-main:
+    name: "Deploy main"
+    if: github.ref == 'refs/heads/main'
+    uses: ./.github/workflows/deploy_tre_reusable.yml
+    with:
+      ciGitRef: ${{ github.ref }}
+      e2eTestsCustomSelector: "extended or shared_services or extended_aad"
```

Comment:
This is the only change, right?
Any special reason to split this into its own workflow rather than use an expression for the different event types?

Reply:
No special reason really, I'll look into using event types, thanks!